proto-java-plugin/src/main/java/io/github/lwlee2608/proto/plugin/DownloadMojo.java

```diff
@@ -13,9 +13,20 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.Set;
 
 @Mojo(name = "download")
 public class DownloadMojo extends AbstractMojo {
+
```

Comment:
We intentionally didn't want to expose `url` as an option. But I guess it should be ok.. @JckHoe what do you think?

Reply:
Yeah this is ok as its the artifact repo url

Yeah was removed previously because the url config was the whole path which allows the user to download anything.

I think this is nice, this will allow the plugin to be overridden to point to artifactory that is hosted privately.

Reply:
good to be able to switch mirrors if one of it is slow 

managed to encounter this scenario

![linux_stuck](https://user-images.githubusercontent.com/18339153/232089771-846436a8-5a3a-4c73-b891-3e1a28d9a2d1.gif)

`https://repo1.maven.org/maven2` was slow and took 50 seconds to complete the above run. 

where after switching to one of the suggested mirrors `https://repo.maven.apache.org/maven2` the download was fast again.

Reply:
I think the repo may throttle the speed if download too frequent? It took me more than 1 hour when I did the testing.

Reply:
@hsrky @JckHoe I agree this is not ideal. We have to redownload `protoc` every time we do a `mvn clean install`. Perhaps we can grab the protoc from local .m2 cache?

Reply:
its possible but the code will be more complex as it needs maven-core as dependency, perhaps lets see how the code looks like when its done, I'll try in on a separate branch and update you guys